src/scripts/choices.test.ts

```diff
@@ -2161,6 +2161,24 @@ describe('choices', () => {
       });
     });
 
+    describe('_onKeyUp', () => {
+      it('triggers a changeQuery event on the passed element', done => {
+        instance.input.value = 'foobar';
+        passedElement.addEventListener(
+          'changeQuery',
+          event => {
+            expect(event.detail).to.eql({
+              value: 'foobar',
+            });
+            done();
+          },
+          false,
+        );
+
```

Comment:
Let's also add a check to ensure that `changeQuery` is fired before the `search` event. You can see an example of firing the `search` event [here](https://github.com/Choices-js/Choices/commit/9605f367f9021324f5d03023005be1086ace9954#diff-12fe009a4650f4b3956664b92225c8a78cd2fc3c9fd22667a0161131062c130dR1739-R1766).